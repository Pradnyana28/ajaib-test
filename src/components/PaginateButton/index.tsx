import React from 'react';
import PageButton from '../PageButton';

const PaginateButton = (props: {
    page: number;
    totalPage: number;
    handlePaginateButton: (pageNumber: number) => void;
}) => {
    return (
        <>
            {Array.from({ length: props.totalPage }, (v, i) => i + 1).map((pageNumber) => {
                if (props.totalPage > 10) {
                    if (pageNumber <= 6 || pageNumber >= props.totalPage - 4) {
                        return <PageButton key={pageNumber} page={props.page} pageNumber={pageNumber} handlePaginateButton={props.handlePaginateButton} />;
                    }

                    if (pageNumber === 7) {
                        return <p>...</p>
                    }
                } else {
                    return <PageButton key={pageNumber} page={props.page} pageNumber={pageNumber} handlePaginateButton={props.handlePaginateButton} />;
                }
            })}
        </>
    );
};

export default PaginateButton;