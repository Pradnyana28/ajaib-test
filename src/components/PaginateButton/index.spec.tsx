import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import PaginateButton from '.';

describe('<PaginateButton />', () => {
    it('renders without crashing', async () => {
        const { container } = render(<PaginateButton page={1} totalPage={10} />);
        expect(container).toMatchSnapshot();
    });
});