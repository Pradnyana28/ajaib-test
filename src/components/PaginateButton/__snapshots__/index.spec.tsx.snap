// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaginateButton /> renders without crashing 1`] = `
<div>
  <button
    class="chakra-button css-4xx2wk"
    data-active=""
    type="button"
  >
    1
  </button>
  <button
    class="chakra-button css-4xx2wk"
    type="button"
  >
    2
  </button>
  <button
    class="chakra-button css-4xx2wk"
    type="button"
  >
    3
  </button>
  <button
    class="chakra-button css-4xx2wk"
    type="button"
  >
    4
  </button>
  <button
    class="chakra-button css-4xx2wk"
    type="button"
  >
    5
  </button>
  <button
    class="chakra-button css-4xx2wk"
    type="button"
  >
    6
  </button>
  <button
    class="chakra-button css-4xx2wk"
    type="button"
  >
    7
  </button>
  <button
    class="chakra-button css-4xx2wk"
    type="button"
  >
    8
  </button>
  <button
    class="chakra-button css-4xx2wk"
    type="button"
  >
    9
  </button>
  <button
    class="chakra-button css-4xx2wk"
    type="button"
  >
    10
  </button>
</div>
`;
